pkg/proxy/service/cluster_service.go

```diff
@@ -65,7 +66,52 @@ func (s *ServiceImpl) GetClusterAccess(userID, username, workspace, proxyPluginN
 		return nil, fmt.Errorf("the requested space is not available")
 	}
 
-	return s.accessForSpace(space, signup.CompliantUsername, proxyPluginName)
+	return s.accessForSpace(space, complaintUserName, proxyPluginName)
+}
+
+func (s *ServiceImpl) getUserSignupComplaintName(userID, username string, publicViewerEnabled bool) (string, error) {
+	// if PublicViewer is enabled and the requested user is the PublicViewer, than no lookup is required
+	if publicViewerEnabled && username == toolchainv1alpha1.KubesawAuthenticatedUsername {
```

Comment:
Will your tests fail if you replace this by
```suggestion
	if username == toolchainv1alpha1.KubesawAuthenticatedUsername {
```
?
If yes then we are good. If not then I would add a test case for this.

Reply:
great point! that should be it: d9ba99c0ac11bcebf5b9c246090ff1e94c11e8dd